modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
+
+		$crypted_id = $this->encrypt($id_order);
+
+		return $crypted_id;
+	}
+
+	/*
+	 * Get the order id using the order uuid
+	 */	
+	public function getOrderIdByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			$first_order = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
 				->orderBy('timestamp', 'desc')
 				->first();
 		} catch (\Exception $e) {
-				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
 		}
 
-		$row_in_array = array(
-			"id" => $existing_order->id,
-			"order_id" => $existing_order->id_order,
-			"timestamp"=> $existing_order->timestamp,
-			"status" => $existing_order->status,
-			"value" => $existing_order->value,
-			"bits" => $existing_order->bits,
-			"bits_payed" => $existing_order->bits_payed
-		);
+		return $first_order->id_order;
+	}
 
-		return $row_in_array;
+	/*
+	 * Get all orders linked to uuid
+	 */	
+	public function getAllOrdersByHash($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		try {
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		} catch (\Exception $e) {
+				echo "Unable to get order id from blockonomics_bitcoin_orders: {$e->getMessage()}";
+		}
+		return;
+	}
+
+	/*
+	 * Get client currency by invoice id
+	 */	
+	public function getClientCurrency($order_hash) {
+		$order_id = $this->decrypt($order_hash);
+		$all_orders_by_id = Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		return $all_orders_by_id;
+	}
+
+	/*
+	 * Check for pending orders and return if exists
+	 */	
+	public function isOrderPending($orders) {
+		foreach ($orders as $order) {
+			//check if status 0 or 1
+			if($order->status == 0 || $order->status == 1){
+				return $order;
+			}
+		}
+		return;
+	}
+
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+	public function isOrderWaiting($orders, $blockonomics_currency) {
+		foreach ($orders as $order) {
+			//check for currency address already waiting
+			if($order->blockonomics_currency == $blockonomics_currency && $order->status == -1){
+				//Check if existing order is expired
+				$current_time = time();
+				$total_time = $this->getTimePeriod() * 60;
+				$clock = $order->timestamp + $total_time - $current_time;
+				if($clock < 0){
+					$order->bits = $this->getNewAmount($order->value, $order->blockonomics_currency);
+					$order->timestamp = $current_time;
+				}
+				$this->updateOrderExpected($order->addr, $order->blockonomics_currency, $order->timestamp, $order->bits);
+				return $order;
+			}
+		}
+		return;
 	}
 
 	/*
-	 * Try to get order row from db by uuid
+	 * Check for unused address or create new
+	 */	
+	public function isOrderNew($orders, $blockonomics_currency) {
```

Comment:
Again choose a bettter function name. _isOrderNew_ is only supposed to do a check request and return a flag.

Reply:
Updated in 18ea106